x-pack/plugins/transform/public/app/sections/transform_management/components/edit_transform_flyout/edit_transform_flyout_form.tsx

```diff
@@ -5,23 +5,52 @@
  * 2.0.
  */
 
-import React, { FC } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 
-import { EuiForm, EuiAccordion, EuiSpacer } from '@elastic/eui';
+import { EuiForm, EuiAccordion, EuiSpacer, EuiSelect, EuiFormRow } from '@elastic/eui';
 
 import { i18n } from '@kbn/i18n';
 
 import { EditTransformFlyoutFormTextInput } from './edit_transform_flyout_form_text_input';
 import { UseEditTransformFlyoutReturnType } from './use_edit_transform_flyout';
+import { useAppDependencies } from '../../../../app_dependencies';
+import { KBN_FIELD_TYPES } from '../../../../../../../../../../../../../../private/var/tmp/_bazel_quynhnguyen/bd5cc7ce3740c1abb2c63a2609d8bb9f/execroot/kibana/bazel-out/darwin-fastbuild/bin/packages/kbn-field-types';
 
 interface EditTransformFlyoutFormProps {
   editTransformFlyout: UseEditTransformFlyoutReturnType;
+  indexPatternId?: string;
 }
 
 export const EditTransformFlyoutForm: FC<EditTransformFlyoutFormProps> = ({
   editTransformFlyout: [state, dispatch],
+  indexPatternId,
 }) => {
   const formFields = state.formFields;
+  const [dateFieldNames, setDateFieldNames] = useState<string[]>([]);
+
+  const appDeps = useAppDependencies();
+  const indexPatternsClient = appDeps.data.indexPatterns;
+
+  useEffect(() => {
+    const getDateFields = async () => {
+      if (indexPatternId !== undefined) {
+        const indexPattern = await indexPatternsClient.get(indexPatternId);
+        if (indexPattern) {
+          const dateTimeFields = indexPattern.fields
+            .filter((f) => f.type === KBN_FIELD_TYPES.DATE)
+            .map((f) => f.name)
+            .sort();
+          setDateFieldNames(dateTimeFields);
+        }
+      }
+    };
+
+    getDateFields();
+  }, [indexPatternId, indexPatternsClient]);
```

Comment:
You have an async logic with set state in effect, hence you should take care of unmounting as well, otherwise you might receive a React error about changing a state on unmounted component. Also, there is no need in defining a function and immediately call, you can try something like this instead 
```suggestion
  useEffect(async function getDateFields() {
      if (indexPatternId !== undefined) {
        const indexPattern = await indexPatternsClient.get(indexPatternId);
        if (indexPattern) {
          const dateTimeFields = indexPattern.fields
            .filter((f) => f.type === KBN_FIELD_TYPES.DATE)
            .map((f) => f.name)
            .sort();
          setDateFieldNames(dateTimeFields);
        }
      }
  }, [indexPatternId, indexPatternsClient]);
```

Reply:
Is mounted example.
https://github.com/elastic/kibana/blob/3bdd764877e708f8cff37d6113dbc618eb6c05ca/x-pack/plugins/ml/public/application/components/saved_objects_warning/saved_objects_warning.tsx#L32

Reply:
Updated here https://github.com/elastic/kibana/pull/108804/commits/27b1c433657f6fa94f35b256c151a3ef138d9863